v1/src/test/java/com/google/cloud/teleport/bigtable/BeamRowToBigtableFnTest.java

```diff
@@ -586,6 +670,375 @@ public void processElementNullColumn() {
+                ParDo.of(
+                    BeamRowToBigtableFn.createWithSplitLargeRows(
+                        ValueProvider.StaticValueProvider.of("#"),
+                        ValueProvider.StaticValueProvider.of("default"),
+                        StaticValueProvider.of(false),
+                        -1,
+                        ValueProvider.StaticValueProvider.of(""),
+                        null)));
+
+    // Setup the expected values and match with returned values.
+    List<Mutation> mutations = new ArrayList<>();
+
+    mutations.add(
+        createMutation(
+            columnFamily,
+            int32ColumnName,
+            ByteString.copyFrom(Bytes.toBytes(int32Value)),
+            bigtableInt32Writetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "mapColumnName[0].key",
+            ByteString.copyFrom(Bytes.toBytes(0)),
+            bigtableMapWritetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "mapColumnName[0].value",
+            ByteString.copyFrom(Bytes.toBytes(mapValue.get(0))),
+            bigtableMapWritetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "mapColumnName[1].key",
+            ByteString.copyFrom(Bytes.toBytes(1)),
+            bigtableMapWritetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "mapColumnName[1].value",
+            ByteString.copyFrom(Bytes.toBytes(mapValue.get(1))),
+            bigtableMapWritetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "mapColumnName[2].key",
+            ByteString.copyFrom(Bytes.toBytes(2)),
+            bigtableMapWritetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "mapColumnName[2].value",
+            ByteString.copyFrom(Bytes.toBytes(mapValue.get(2))),
+            bigtableMapWritetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "listColumnName[0]",
+            ByteString.copyFrom(Bytes.toBytes(listValue.get(0))),
+            bigtableListWritetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "listColumnName[1]",
+            ByteString.copyFrom(Bytes.toBytes(listValue.get(1))),
+            bigtableListWritetime));
+    mutations.add(
+        createMutation(
+            columnFamily,
+            "listColumnName[2]",
+            ByteString.copyFrom(Bytes.toBytes(listValue.get(2))),
+            bigtableListWritetime));
+
+    final List<KV<ByteString, Iterable<Mutation>>> expectedBigtableRows =
+        ImmutableList.of(KV.of(ByteString.copyFrom(Bytes.toBytes("rowkeyvalue")), mutations));
+
+    PAssert.that(bigtableRows).containsInAnyOrder(expectedBigtableRows);
+
+    // Run the pipeline
+    pipeline.run();
+  }
+
+  @Test
+  public void processElementWithBackfillNowTimestamp() {
```

Comment:
Add a test with 0 timestamp
Add a test with timestamp = System.currentTimeInMillis() + 1000 and see what the timestamp comes out in CBT, is it same?

Reply:
Added tests for these scenarios.